src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,34 +1,66 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
+import React, { ChangeEvent } from 'react';
+import { StatusFilter } from '../../helpers/StatusFilter';
 
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
+type Props = {
+  inputValue: string,
+  setInputValue: (input: React.SetStateAction<string>) => void,
+  setTodoFilter: (status: React.SetStateAction<StatusFilter>) => void,
+};
 
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button
-          data-cy="clearSearchButton"
-          type="button"
-          className="delete"
+export const TodoFilter: React.FC<Props> = ({
+  inputValue,
+  setInputValue,
+  setTodoFilter,
+}) => {
+  const handleClearinput = () => {
+    setInputValue('');
+  };
+
+  const changeFilter = (event: ChangeEvent<HTMLSelectElement>) => {
+    setTodoFilter(event.target.value as StatusFilter);
+  };
+
+  return (
+    <form className="field has-addons">
+      <p className="control">
+        <span className="select">
+          <select
+            data-cy="statusSelect"
+            onChange={changeFilter}
+          >
+            <option value={StatusFilter.All}>All</option>
+            <option value={StatusFilter.ACTIVE}>Active</option>
+            <option value={StatusFilter.COMPLETED}>Completed</option>
+          </select>
+        </span>
+      </p>
+
+      <p className="control is-expanded has-icons-left has-icons-right">
+        <input
+          data-cy="searchInput"
+          type="text"
+          className="input"
+          value={inputValue}
+          onChange={(event) => setInputValue(event.target.value)}
```

Comment:
let's create a handler where we put our `event`
```suggestion
          onChange={handleInputChange}
```